week1/homework/test/routes.spec.js

```diff
@@ -0,0 +1,83 @@
+const request = require("supertest");
+const app = require("../src/backend/app");
+const meals = require("../src/backend/data/meals");
+const reviews = require("../src/backend/data/reviews");
+const reservations = require("../src/backend/data/reservations");
+
+describe("GET /", () => {
+    test("responds with corrected response", async () => {
+        const response = await request(app).get("/");
+        expect(response.statusCode).toBe(200);
+        expect(response.text).toBe("Meal Sharing Web App");
+    });
+});
+
+describe("GET /meals", () => {
+    test("responds with all meals (with their reviews, if any) from data file", async () => {
+        // Deep copy meals
+        const copiedMeals = JSON.parse(JSON.stringify(meals));
+        const mealsWithReviews = copiedMeals.map(meal => {
+            meal.reviews = reviews.filter(review => review.mealId === meal.id);
+            return meal;
+        });
+        const response = await request(app).get("/meals");
+        expect(response.statusCode).toBe(200);
+        expect(response.body).toMatchObject(mealsWithReviews);
+        expect(Array.isArray(response.body)).toBeTruthy();
+    });
+});
+
+describe("GET /cheap-meals", () => {
+    test("route exists", async () => {
+        const response = await request(app).get("/cheap-meals");
+        expect(response.statusCode).toBe(200);
```

Comment:
Just ensure a route exists and an array is returned, since determination of this and `large-meals` is arbitrary.